mimic/rest/identity_api.py

```diff
@@ -669,6 +688,521 @@ def list_tenants(self, request):
                 'message': ("No valid token provided. Please use the 'X-Auth-Token'"
                             " header with a valid token.")}})
 
+    @app.route('/v2.0/services', methods=['GET'])
+    def list_external_api_services(self, request):
+        """
+        List the available external services that endpoint templates
+        may be added to.
+
+        .. note:: Part of the OS-KSADM extension.
+        .. note:: Does not implement the limits or markers.
+        """
+        x_auth_token = request.getHeader(b"x-auth-token")
```

Comment:
This `x_auth_token` check ought to be a function, or better yet, a decorator that declares authorization is required.  Although mimic is not exactly security sensitive currently, duplicating this kind of code is a very risky anti-pattern; since mimic *may* eventually be handling administrative credentials (and this change only makes it more likely that it will be) we should treat it as though it were.  Please de-duplicate this with the other auth-token checks so that if it does more than check for the existence of the header in the future we don't have to re-implement it.